BrainPortal/app/models/cluster_task.rb

```diff
@@ -329,6 +329,26 @@ def safe_userfile_find_or_new(klass,attlist)
     return klass.new(attlist)
   end
 
+  # Returns whether or not the specified userfile exists
+  # Should at least specify the :name and :data_provider_id attributes.
+  # But the latter is given a default value if not.
+  def userfile_exists(klass,attlist)
```

Comment:
Someone coudl wonder why this method is part of the CbrainTask::ClusterTask class; add in the documentation that it's meant to help task developers create safe code for creating output files, by filling many of the attributes of the userfiles with defaults taken from the task itself (e.g. task's owner, task's group, task's results_data_provider_id etc). The method above, safe_userfile_find_or_new did exactly this.